### ISSUE TITLE
Add custom bounties to Bountiful's board

### ISSUE THREAD

**Issue by DoctorScorch** [NillerMedDild/Enigmatica6#3732]
### Describe your suggestion

Since Bountiful is already in the pack, how about adding some custom bounties with custom rewards through it's datapack editing?

Rewards like Tetra's schematics, trinkets, curios, materials for tools, Apotheosis' affix books or items, ingredients for magic feather or other materials that are difficult to find due to exploring would be neat, since the bounties could be used as an minor alternative way to "come across them". Maybe some QoL items or others that make the grind on expert mode easier could also be used as rewards

Design-wise it does crosses a little with the quests from the pack but since the bounties could be random and timesensitive, as well as the rewards being able to go on a different direction compared to those of the questbook could make it worthwhile

### Why would you like this added/changed?

It certainly looks like a neat thing to add to the pack at first glance to give it one more layer of detail (as well as another TIP for the loading screen :D) due to it's design space
In short, https://www.youtube.com/watch?v=Y0PKG5-t3zU

**Comment by NillerMedDild** [comment]
Good suggestion, I'm all for it! Someone just has to actually implement it, I don't really have the time 😅

**Comment by MuteTiefling** [comment]
I don't mind coding them, but I have very few ideas here. Anyone willing to write up detailed entries?

**Comment by DoctorScorch** [comment]
Well, I don't know if I can help much on that but I was reading a little about how the mod works and it's functionality seems to be mainly tied to pools, weights, rarities and a combination of those, so... rather than making bounties that work like "Kill X, get Y" the mod seems to function as "This Bounty Board has Decree Nº1 on one of it's three slots, so it draws from X and Y reward Pool and Z objective Pool" and then the bounties are randomly generated over time, tying together some objectives and some rewards according to their value.

Speaking about objectives, they could be met either by killing a certain number of mobs, for example 1x of a boss from Meet Your Fight, 6x of an Undergarden mob, 8x Wraiths or something similar, or by delivering some items like X amount of Neptunium or Y amount of Living Rock. The main factor is to define the value of those objectives and what pools they should be.

Meanwhile rewards work in a very similar way, except they have the limitation of only being capable of giving items, like 1x Expertise Treatise, 1x Running Shoes, 2x Pristine Diamonds, 8x Pristine Emeralds and have a rarity system tied to them

Hopefully those examples can give someone an idea on how to better implement this feature (I have put a more general idea of other things that could be used as reward on the first comment of this issue). I wish I could help more with some suggestions of the values, but idk enough about the intricacies of how the pack is structured (especially the expert one) so as to not break it or make the bounties by mistake too hard to be fulfilled till late game xD

**Comment by DoctorScorch** [comment]
Ok, enough with the walls of text for today, me 😅

**Comment by werwerwsx** [comment]
What I'm concerned about is the decree's seem to be hard coded with no way to add new ones (that I can find, if I'm wrong I apologize). Which would make it difficult to separate the pools of rewards and quests.

**Comment by TemperedFool** [comment]
> I don't mind coding them, but I have very few ideas here. Anyone willing to write up detailed entries? 

@MuteTiefling does it work via loot tables? If so maybe we can leverage the questing loot box tables for rewards? Require a relatively high tier pool of things from a mod to get some items from one of its tables?

**Comment by DoctorScorch** [comment]
Well, there’s a 1.14 Github page on how it implements it’s functions though datapacks and how to edit them. Maybe it’s similar for 1.16?
https://github.com/ejektaflex/Bountiful/wiki/1.14+%20Data-Packs

**Comment by MuteTiefling** [comment]
Looks like it uses it's own system then, rather than loot tables. Bummer. Would have been an easy win there.
